src/main/webapp/app/core/course/overview/course-overview.component.ts

```diff
@@ -98,402 +53,216 @@ import { facSidebar } from 'app/shared/icons/icons';
-                        courses.push(course);
-                    });
-                    this.courses = sortCourses(courses);
-                    if (this.courses.length > this.MIN_DISPLAYED_COURSES) {
-                        const lastAccessedCourseIds = this.courseAccessStorageService.getLastAccessedCourses(CourseAccessStorageService.STORAGE_KEY_DROPDOWN);
-                        this.courses = this.courses.filter((course) => lastAccessedCourseIds.includes(course.id!));
-                    }
-                    this.courses = this.courses.filter((course) => course.id !== this.courseId);
+                    this.course.set(res.body);
                 }
-            },
-        });
+
+                this.setUpConversationService();
+
+                setTimeout(() => this.refreshingCourse.set(false), 500); // ensure min animation duration
+            }),
+            // catch 403 errors where registration is possible
+            catchError((error: HttpErrorResponse) => {
+                if (error.status === 403) {
+                    this.redirectToCourseRegistrationPageIfCanRegisterOrElseThrow(error);
+                    // Emit a default value, for example `undefined`
+                    return of(undefined);
+                } else {
+                    return throwError(() => error);
+                }
+            }),
+            // handle other errors
+            catchError((error: HttpErrorResponse) => {
+                const errorMessage = error.headers.get('X-artemisApp-message')!;
+                this.alertService.addAlert({
+                    type: AlertType.DANGER,
+                    message: errorMessage,
+                    disableTranslation: true,
+                });
+                return throwError(() => error);
+            }),
+        );
+        // Start fetching, even if we don't subscribe to the result.
+        // This enables just calling this method to refresh the course, without subscribing to it:
+        this.loadCourseSubscription?.unsubscribe();
+        if (refresh) {
+            this.loadCourseSubscription = observable.subscribe();
+        }
+        return observable;
     }
 
-    /** Navigate to a new Course */
-    switchCourse(course: Course) {
-        this.router.navigateByUrl('/', { skipLocationChange: true }).then(() => {
-            this.router.navigate(['courses', course.id, 'exercises']);
-        });
+    protected getHasSidebar(): boolean {
+        return !!this.route.snapshot.firstChild?.data?.hasSidebar;
     }
 
-    getCourseActionItems(): CourseActionItem[] {
-        const courseActionItems = [];
-        this.canUnenroll = this.canStudentUnenroll() && !this.course?.isAtLeastTutor;
-        if (this.canUnenroll) {
-            const unenrollItem: CourseActionItem = this.getUnenrollItem();
-            courseActionItems.push(unenrollItem);
+    protected handleComponentActivation(componentRef: any): void {
+        if (
```

Comment:
just out of curiosity, it's really minor: can we rather tell the method if `activatedComponentReference` should be set?

this would - even though the dependency seems non-critical - make the overview component more independent of the caller.